esi_leap/objects/offer.py

```diff
@@ -63,9 +65,14 @@ def get_availabilities(self):
             return []
 
         conflicts = self.dbapi.offer_get_conflict_times(self)
-
+        now = datetime.datetime.now()
         if conflicts:
-            a = [self.start_time, conflicts[0][0]]
+            if now <= self.start_time:
+                a = [self.start_time, conflicts[0][0]]
+            elif now < conflicts[0][0]:
+                a = [now, conflicts[0][0]]
+            else:
+                a = []
```

Comment:
Not your fault, but I'm having trouble understanding the logic in this entire function. Could you add some comments explaining?